### PR TITLE
project_closing will close the related analytic account when a project i...

### PR DESCRIPTION
**Pull request by vrenaville**
project_closing will close the related analytic account when a project is closed
and reopen it if the project is reopen

### CODE REVIEW

**Comment by dreispt** [comment]
@vrenaville I'm sorry, but do you mind rebasing against current 7.0 branch?
 #32 has just been merged and it helps a lot for the review process.

**Comment by vrenaville** [comment]
@dreispt Thanks for the review, I have just rebase my branch

Vincent

**Comment by vrenaville** [comment]
Hum Travis failed due to project_action_item dependencies

File: project_closing/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#
+##############################################################################
+{
+    "name": "Project closing",
+    "version": "1.1",
+    "author": "Camptocamp",
+    "website": "http://www.camptocamp.com",
+    "category": "project Management",
+    "depends": ["project"],
+    "description": """\
```
Comment (pedrobaeza) [inline project_closing/__openerp__.py]: Remove \

File: project_closing/project.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from osv import orm
+
+
+class project_project(orm.Model):
+    _inherit = 'project.project'
+
+    def set_done(self, cr, uid, ids, context=None):
+        ''' We will close related analytic account '''
```
Comment (pedrobaeza) [inline project_closing/project.py]: Use """ for docstring.

File: project_closing/project.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        projects = self.browse(cr, uid, ids, context=context)
+        for project in projects:
+            analytic_account_obj.write(cr, uid,
+                                       [project.analytic_account_id.id],
+                                       {'state': 'close'}, context=context)
+        return super(project_project, self).set_done(cr, uid,
+                                                     ids, context=context)
+
+    def set_open(self, cr, uid, ids, context=None):
+        ''' 'We will re-open related analytic account '''
```
Comment (pedrobaeza) [inline project_closing/project.py]: Use """ for docstring.

**Comment by dreispt** [comment]
@pedrobaeza I don't see the problem with the lines you pointed out. Can you elaborate?

**Comment by pedrobaeza** [comment]
According PEP 257 docstrings, triple quota must be with " (http://legacy.python.org/dev/peps/pep-0257/#multi-line-docstrings).

About the backslash, inside a multi-line text doesn't do anything, except scaping following character.

**Comment by dreispt** [comment]
On backslash: yes it does - it avoids the top empty line from the string.

```
>>> """\
... 1
... 2
... """
'1\n2\n'
>>> """
... 1
... 2
... """
'\n1\n2\n'
```

**Comment by pedrobaeza** [comment]
The link was not accurate. I quote the text that it's upper on the link:

```
For consistency, always use """triple double quotes""" around docstrings. Use r"""raw triple double quotes""" if you use any backslashes in your docstrings. For Unicode docstrings, use u"""Unicode triple-quoted strings""".
```

About description, it's parsed as RST, so the backslash is ignored I think, but I'm not going to block for that.

File: project_closing/project.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    def set_done(self, cr, uid, ids, context=None):
+        ''' We will close related analytic account '''
+        analytic_account_obj = self.pool.get('account.analytic.account')
+        if isinstance(ids, (int, long)):
+            ids = [ids]
+        projects = self.browse(cr, uid, ids, context=context)
+        for project in projects:
+            analytic_account_obj.write(cr, uid,
+                                       [project.analytic_account_id.id],
+                                       {'state': 'close'}, context=context)
```
Comment (dreispt) [inline project_closing/project.py:18]: Suggestion (more efficient - one write only):

```
project_ids = [project.analytic_account_id.id for project in projects]
analytic_account_obj.write(cr, uid, project_ids, {'state': 'close'}, context=context)
```

**Comment by vrenaville** [comment]
Thanks for the review
@pedrobaeza @dreispt 
Hello, I have just improve my PR according to your comments.

Vincent

**Comment by pedrobaeza** [comment]
Travis errors are unrelated.

:+1:

**Comment by guewen** [comment]
:+1:

**Pull request merged**